src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs

```diff
@@ -143,11 +174,15 @@ protected static async Task PrecalculateAsync(Document document, IndexCreator cr
             {
                 var storage = await persistentStorageService.GetStorageAsync(SolutionKey.ToSolutionKey(solution), cancellationToken).ConfigureAwait(false);
                 await using var _ = storage.ConfigureAwait(false);
-                // Check if we've already stored a checksum and it matches the checksum we 
-                // expect.  If so, we're already precalculated and don't have to recompute
-                // this index.  Otherwise if we don't have a checksum, or the checksums don't
-                // match, go ahead and recompute it.
-                return await storage.ChecksumMatchesAsync(document, s_persistenceName, checksum, cancellationToken).ConfigureAwait(false);
+
+                // Check if we've already stored a checksum and it matches the checksum we expect.  If so, we're already
+                // precalculated and don't have to recompute this index.  Otherwise if we don't have a checksum, or the
+                // checksums don't match, go ahead and recompute it.
+                //
+                // Check with both checksums as we don't know at this reading point if the document has pp-directives in
+                // it or not, and we don't want parse the document to find out.
```

Comment:
Clever that this works -- implicitly the "did we have a preprocessor directive or not?" is just which checksum matches.